dmipy/core/modeling_framework.py

```diff
@@ -2191,6 +2193,383 @@ def __call__(self, acquisition_scheme, **kwargs):
+        that are outside the brain). If no mask is given then all voxels are
+        included.
+
+        An optimization approach can be chosen as either 'brute2fine' or 'mix'.
+        - Choosing brute2fine will first use a brute-force optimization to find
+          an initial guess for parameters without one, and will then refine the
+          result using gradient-descent-based optimization.
+
+          Note that given no initial guess will make brute2fine precompute an
+          global parameter grid that will be re-used for all voxels, which in
+          many cases is much faster than giving voxel-varying initial condition
+          that requires a grid to be estimated per voxel.
+
+        - Choosing mix will use the recent MIX algorithm based on separation of
+          linear and non-linear parameters. MIX first uses a stochastic
+          algorithm to find the non-linear parameters (non-volume fractions),
+          then estimates the volume fractions while fixing the estimates of the
+          non-linear parameters, and then finally refines the solution using
+          a gradient-descent-based algorithm.
+
+        The fitting process can be readily parallelized using the optional
+        "pathos" package. If it is installed then it will automatically use it,
+        but it can be turned off by setting use_parallel_processing=False. The
+        algorithm will automatically use all cores in the machine, unless
+        otherwise specified in number_of_processors.
+
+        Data with multiple TE are normalized in separate segments using the
+        b0-values according that TE.
+
+        Parameters
+        ----------
+        acquisition_scheme : DmipyAcquisitionScheme instance,
+            An acquisition scheme that has been instantiated using dMipy.
+        data : N-dimensional array of size (N_x, N_y, ..., N_dwis),
+            The measured DWI signal attenuation array of either a single voxel
+            or an N-dimensional dataset.
+        mask : (N-1)-dimensional integer/boolean array of size (N_x, N_y, ...),
+            Optional mask of voxels to be included in the optimization.
+        solver : string,
+            TODO: ?
+        Ns : integer,
+            for brute optimization, decised how many steps are sampled for
+            every parameter. TODO: ?
+        maxiter : integer,
+            for MIX optimization, how many iterations are allowed. TODO: ?
+        N_sphere_samples : integer,
+            for brute optimization, how many spherical orientations are sampled
+            for 'mu'. TODO: ?
+        use_parallel_processing : bool,
+            whether or not to use parallel processing using pathos.
+        number_of_processors : integer,
+            number of processors to use for parallel processing. Defaults to
+            the number of processors in the computer according to cpu_count().
+        directions_for_lut : int or N-by-3 array
```

Comment:
I am confused, is the LUT part of the first MR or not?

Reply:
we agreed on _not_